src/runtime/numerics.rs

```diff
@@ -1,1053 +1,819 @@
     fn div(self, rhs: Vector2) -> Vector2 {
-        Vector2 {
-            x: self.x / rhs.x,
-            y: self.y / rhs.y,
-        }
+        self.impl_div(&rhs)
     }
 }
 impl ::std::ops::Div<&Vector2> for Vector2 {
     type Output = Vector2;
     fn div(self, rhs: &Vector2) -> Vector2 {
-        Vector2 {
-            x: self.x / rhs.x,
-            y: self.y / rhs.y,
-        }
+        self.impl_div(rhs)
     }
 }
 impl ::std::ops::Div<Vector2> for &Vector2 {
     type Output = Vector2;
     fn div(self, rhs: Vector2) -> Vector2 {
-        Vector2 {
-            x: self.x / rhs.x,
-            y: self.y / rhs.y,
-        }
+        self.impl_div(&rhs)
     }
 }
 impl ::std::ops::Div<&Vector2> for &Vector2 {
     type Output = Vector2;
     fn div(self, rhs: &Vector2) -> Vector2 {
-        Vector2 {
-            x: self.x / rhs.x,
-            y: self.y / rhs.y,
-        }
+        self.impl_div(rhs)
     }
 }
 impl ::std::ops::Div<f32> for Vector2 {
     type Output = Vector2;
     fn div(self, rhs: f32) -> Vector2 {
-        Vector2 {
-            x: self.x / rhs,
-            y: self.y / rhs,
-        }
+        self.impl_div_f32(rhs)
     }
 }
 impl ::std::ops::Div<f32> for &Vector2 {
     type Output = Vector2;
     fn div(self, rhs: f32) -> Vector2 {
-        Vector2 {
-            x: self.x / rhs,
-            y: self.y / rhs,
-        }
+        self.impl_div_f32(rhs)
     }
 }
 impl ::std::ops::Mul<Vector2> for Vector2 {
     type Output = Vector2;
     fn mul(self, rhs: Vector2) -> Vector2 {
-        Vector2 {
-            x: self.x * rhs.x,
-            y: self.y * rhs.y,
-        }
+        self.impl_mul(&rhs)
     }
 }
 impl ::std::ops::Mul<&Vector2> for Vector2 {
     type Output = Vector2;
     fn mul(self, rhs: &Vector2) -> Vector2 {
-        Vector2 {
-            x: self.x * rhs.x,
-            y: self.y * rhs.y,
-        }
+        self.impl_mul(rhs)
     }
 }
 impl ::std::ops::Mul<Vector2> for &Vector2 {
     type Output = Vector2;
     fn mul(self, rhs: Vector2) -> Vector2 {
-        Vector2 {
-            x: self.x * rhs.x,
-            y: self.y * rhs.y,
-        }
+        self.impl_mul(&rhs)
     }
 }
 impl ::std::ops::Mul<&Vector2> for &Vector2 {
     type Output = Vector2;
     fn mul(self, rhs: &Vector2) -> Vector2 {
-        Vector2 {
-            x: self.x * rhs.x,
-            y: self.y * rhs.y,
-        }
+        self.impl_mul(rhs)
     }
 }
 impl ::std::ops::Mul<f32> for Vector2 {
     type Output = Vector2;
     fn mul(self, rhs: f32) -> Vector2 {
-        Vector2 {
-            x: self.x * rhs,
-            y: self.y * rhs,
-        }
+        self.impl_mul_f32(rhs)
     }
 }
 impl ::std::ops::Mul<f32> for &Vector2 {
     type Output = Vector2;
     fn mul(self, rhs: f32) -> Vector2 {
-        Vector2 {
-            x: self.x * rhs,
-            y: self.y * rhs,
-        }
+        self.impl_mul_f32(rhs)
     }
 }
-impl Vector2 {
+
+// Vector3
+impl Vector3 {
     pub fn zero() -> Self {
```

Comment:
These might be better suited as constants.

Reply:
Yes, good idea.